lib/ui/window/platform_isolate.cc

```diff
@@ -0,0 +1,47 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include <iostream>
+#include <thread>
+
+#include "flutter/fml/macros.h"
+#include "flutter/fml/task_runner.h"
+#include "flutter/lib/ui/ui_dart_state.h"
+#include "flutter/lib/ui/window/platform_isolate.h"
+#include "third_party/tonic/converter/dart_converter.h"
+
+namespace flutter {
+
+void PlatformIsolateNativeApi::Spawn(Dart_Handle entry_point) {
+  UIDartState* current_state = UIDartState::Current();
+  FML_DCHECK(current_state != nullptr);
+  if (!current_state->IsRootIsolate()) {
+    // TODO(flutter/flutter#136314): Remove this restriction.
+    Dart_EnterScope();
+    Dart_ThrowException(tonic::ToDart(
+        "PlatformIsolates can only be spawned on the root isolate."));
+  }
+
```

Comment:
Jason mentioned the current PR may not work on desktop embedders (which may have their own platform thread event loop). If so, can we detect this here and throw? (ideally also test)

Reply:
The Flutter engine does not currently have a way to detect whether the embedder is using a custom platform thread task runner that does not support this usage.

After this lands I can follow up with another PR that adds an engine flag which will disable the platform isolate feature.  If the embedder sets the flag, then this API will throw.